app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/plugins/GoogleSheetsPlugin.java

```
@@ -81,6 +81,23 @@ public Mono<ActionExecutionResult> executeParameterized(
                 ExecuteActionDTO executeActionDTO,
                 DatasourceConfiguration datasourceConfiguration,
                 ActionConfiguration actionConfiguration) {
+            return executeParameterizedWithFlags(
+                    connection, executeActionDTO, datasourceConfiguration, actionConfiguration, null);
+        }
+
+        @Override
+        public Mono<TriggerResultDTO> trigger(
+                Void connection, DatasourceConfiguration datasourceConfiguration, TriggerRequestDTO request) {
+            return triggerWithFlags(connection, datasourceConfiguration, request, null);
```

Should we throw unsupported method exception here considering we will always be using the method with feature flag param? Are there any flows where we are still using this method?

---

This is being used in DatasourceTriggerSolution test cases and I feel instead of saying unsupported method, we can fall back to the default not supported version of drive, hence added this method here, let me know if that sounds good

---

We should generally rely on fail early, which provides you the correct code path where the issue is occurring. I was trying to understand if we are in any situation going to hit this method in GSheets plugin, ideally if not we may have to trace manually to get to the correct trace instead of relying on the stack-trace we get by default from throwing exception.

---

Makes sense, this is not explicitly being called for gsheetplugin, it's only in the test cases and to handle test cases I have made this fallback to default